src/ir/proto_to_ir.h

```diff
@@ -33,6 +34,20 @@ class ProtoToIR {
     std::unique_ptr<Module> module;
     std::unique_ptr<SsaNames> ssa_names;
   };
+
+  static Result Convert(IRContext& context,
+                        const std::string irTranslationUnitJsonProto) {
```

Comment:
Thanks for the tip. My habit in other languages is towards immutability for everything, since mutability has usually been a source of bugs, I didn't even realize this was inhibiting compiler opts.

Reply:
Very sympathetic to that, I used to make all of my C++ locals `const`. Unfortunately, when C++ introduced move constructors, they couldn't do what Rust did and enforce that the name was no longer used after the move, as it was incompatible with past approaches, so move constructors are allowed to modify the original object and leave it in a "valid but unspecified state". Which is basically saying "it's no longer live, but we aren't going to stop you from touching it". Very annoying, but the additional freedom to steal resources from expiring objects is quite handy.